packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -56,6 +56,7 @@ export interface TCGCContext {
   previewStringRegex: RegExp;
   disableUsageAccessPropagationToBase: boolean;
   flattenUnionAsEnum?: boolean;
```

Comment:
JS would prefer to have discriminatorProperty for sub model also, could we add that support?  https://github.com/Azure/typespec-azure/blob/b02fc0b5a642c82018ae431da1ba9865a99e8c2e/packages/typespec-client-generator-core/src/interfaces.ts#L504

Reply:
Added.